transformer_engine/pytorch/tensor/quantized_tensor.py

```diff
@@ -154,6 +154,45 @@ class Quantizer(abc.ABC):
     """
     columnwise_usage: bool
 
+    def __init__(self, *, rowwise: bool, columnwise: bool) -> None:
+        self.rowwise_usage = rowwise
+        self.columnwise_usage = columnwise
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}("
+            f"rowwise_usage={self.rowwise_usage}, "
+            f"columnwise_usage={self.columnwise_usage}"
+            ")"
+        )
+
+    @abc.abstractmethod
+    def quantize(self, tensor: torch.Tensor, **kwargs) -> QuantizedTensorBase:
+        """Quantize tensor"""
+
+    def __call__(self, tensor: torch.Tensor) -> QuantizedTensorBase:
+        """Quantize tensor"""
+        return self.quantize(tensor)
+
+    def set_usage(
+        self,
+        rowwise: Optional[bool] = None,
+        columnwise: Optional[bool] = None,
+    ) -> None:
+        """Set how the quantized tensor is expected to be used"""
+        if rowwise is not None:
+            self.rowwise_usage = rowwise
+        if columnwise is not None:
+            self.columnwise_usage = columnwise
+
+
+class Quantizer(QuantizerBase):
```

Comment:
I don't think the distinction between `Quantizer` and `QuantizerBase` is logical. This PR is trying to distinguish between quantizers that call `tex.quantize` and those that call a Python impl, but that's a quantizer-specific implementation detail. `QuantizerBase` is also haphazardly removing parts of the quantizer API, like the ability to construct empty tensors, PyTorch autograd support, etc.

I think the right design is not to add an unnecessary `QuantizerBase` class, but to decouple `Quantizer` from `tex.quantize`. We can add an abstract `quantize_impl` function that is called within `quantize`. The existing quantizers should call `tex.quantize`, but future quantizers could use a pure Python impl.